### ISSUE TITLE
[Strange application] Very Small nanoVLM Configuration – Any Similar Experiences?

### ISSUE THREAD

**Issue by sbrzz** [huggingface/nanoVLM#141]
Hi all,

First of all, congratulations to the nanoVLM team — this project is both impressive and meaningful.

I wanted to share an application I'm working on, to see if anyone has experienced something similar or can offer advice.

I'm using nanoVLM with a significantly reduced model size. Specifically, I modified the VLMConfig as follows:

```python
vit_n_heads: int = 1
vit_n_blocks: int = 1
lm_hidden_dim: int = 72
lm_inter_dim: int = 192
lm_n_heads: int = 1
lm_n_kv_heads: int = 1
lm_n_blocks: int = 1

```

Vision encoder parameters: 7.8M
LLM decoder parameters: 3.6M

I'm training on a private dataset (~4GB, ~3000 samples), where each sample includes **a fixed question** and varying images. The target answers (assistant output) can be relatively long — around 10 to 50 words, which is longer than most examples in the Cauldron dataset.

While the loss decreases during training, after a few epochs the model starts to overfit badly. It ends up generating the same nonsensical output repeatedly.

Has anyone encountered similar behavior with very small VLM configurations? Any suggestions or best practices to improve generalization would be appreciated.

Thanks!

**Comment by andimarafioti** [comment]
Hi! For 3000 samples, I would expect this behavior with the larger model as well. It's not a lot of data, so it makes sense that the model learns to overfit to it. Even if you have 10 to 50 words per answer, that is still only 3000 answers it needs to memorize, which isn't so much. It's actually a good signal if your model overfits to a small dataset.

**Comment by sbrzz** [comment]
Just to say I am working on a blog post to share details on how I managed this situation

**Comment by aidansmyth95** [comment]
Hi Simone,

Very cool article! Looking to do a similar thing but on a different dataset (industrial application on an edge device for IoT).

Can you please share the config you used to train this tiny model? Would help nudge my hyper parameter tuning in the right direction! Also did you use the latest version in the repo (v0.2) or the older version (v0.1)?

Kind regards,
Aidan

**Comment by sbrzz** [comment]
Just to let the community know, the conversation between @aidansmyth95 and me is continuing by email.

If anyone is interested in the details or simply curious, don’t hesitate to contact me.

To the maintainers: I’d like to kindly ask you to reconsider the proposal for a dedicated page where examples of applications can be collected and tracked. I believe it would be very useful for the community.

@lusxvr
